nemo/tron/init.py

```diff
@@ -30,16 +30,24 @@
 )
 from megatron.core.utils import get_te_version, is_te_min_version, is_torch_min_version
 
-from nemo.tron.config import ConfigContainer, RerunStateMachineConfig
-from nemo.tron.state import GlobalState
+from nemo.collections.llm.gpt.model.base import GPTConfig
+from nemo.collections.llm.t5.model.t5 import T5Config
+from nemo.tron.config import DistributedInitConfig, RerunStateMachineConfig, RNGConfig
 from nemo.tron.utils.common_utils import get_local_rank_preinit, get_rank_safe, get_world_size_safe
 
 
 def initialize_megatron(
-    # extra_args_provider=None,
-    # args_defaults={},
-    # ignore_unknown_args=False,
-    cfg: ConfigContainer,
+    model_config: GPTConfig | T5Config,
```

Comment:
this is likely to keep changing with new knobs added, and each required arg would be considered a breaking API change. while the changes to `setup_optimizer` are much more self contained, `initialize_megatron` could get really unwieldy over time. do you think its reasonable to accept the top-level config here?